### ISSUE TITLE
Postgres - Found unparsable section: ALTER INDEX...

### ISSUE THREAD

**Issue by PawelLipski** [sqlfluff/sqlfluff#2978]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Apparent lack of support for `ALTER INDEX` syntax in Postgres dialect :///

File:
```sql
;;; (9 lines redacted)
ALTER INDEX "AKKA_PROJECTION_OFFSET_STORE_pkey" RENAME TO "akka_projection_offset_store_pkey";
ALTER INDEX "PROJECTION_NAME_INDEX" RENAME TO "projection_name_index";
```

### Expected Behaviour

Linter doesn't raise `PRS: Found unparsable section: 'ALTER INDEX...`

### Observed Behaviour

```
==== sqlfluff ====
sqlfluff:               0.11.2 python:                 3.10.3
implementation:        cpython dialect:              postgres
verbosity:                   2 templater:               jinja

== Raw Config:
core:                                   
    dialect:            postgres            
    disable_noqa:       False               
    encoding:           autodetect          
    exclude_rules:      
L001,
L003,
L009,
L010,
L016,
L017,
L039,
L040,
L050,
L059
    fix_even_unparsable:False               
    ignore:             []                  
    ignore_templated_areas:True                
    nocolor:            False               
    output_line_length: 80                  
    recurse:            True                
    runaway_limit:      10                  
    sql_file_exts:      .sql,.sql.j2,.dml,.ddl
    templater:          jinja               
    verbose:            2                   
indentation:                            
    indented_ctes:      False               
    indented_joins:     False               
    indented_using_on:  True                
    template_blocks_indent:True                
rules:                                  
    allow_scalar:       True                
    comma_style:        trailing            
    indent_unit:        space               
    max_line_length:    80                  
    single_table_references:consistent          
    tab_space_size:     4                   
    unquoted_identifiers_policy:all                 
    L007:                                   
        operator_new_lines: after               
    L010:                                   
        capitalisation_policy:consistent          
    L011:                                   
        aliasing:           explicit            
    L012:                                   
        aliasing:           explicit            
    L014:                                   
        extended_capitalisation_policy:consistent          
    L016:                                   
        ignore_comment_clauses:False               
        ignore_comment_lines:False               
    L026:                                   
        force_enable:       False               
    L028:                                   
        force_enable:       False               
    L029:                                   
        unquoted_identifiers_policy:aliases             
    L030:                                   
        extended_capitalisation_policy:consistent          
    L031:                                   
        force_enable:       False               
    L038:                                   
        select_clause_trailing_comma:forbid              
    L040:                                   
        capitalisation_policy:consistent          
    L042:                                   
        forbid_subquery_in: join                
    L047:                                   
        prefer_count_0:     False               
        prefer_count_1:     False               
    L051:                                   
        fully_qualify_join_types:inner               
    L052:                                   
        multiline_newline:  False               
        require_final_semicolon:False               
    L054:                                   
        group_by_and_order_by_style:consistent          
    L057:                                   
        additional_allowed_characters:""                  
        allow_space_in_identifier:False               
        quoted_identifiers_policy:all                 
        unquoted_identifiers_policy:all                 
    L059:                                   
        prefer_quoted_identifiers:False               
    L062:                                   
templater:                              
    unwrap_wrapped_queries:True                
    jinja:                                  
        apply_dbt_builtins: True                
        macros:                                 
            dbt_config:         {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
            dbt_is_incremental: {% macro is_incremental() %}True{% endmacro %}
            dbt_ref:            {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
            dbt_source:         {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
            dbt_var:            {% macro var(variable, default='') %}item{% endmacro %}

==== readout ====

=== [ path: <redacted>.sql ] ===

== [<redacted>.sql] LINTING
== [<redacted>.sql] FAIL
L:  10 | P:   1 |  PRS | Line 10, Position 1: Found unparsable section: 'ALTER
                       | INDEX "AKKA_PROJECTION_OFFSET_STOR...'
L:  11 | P:   1 |  PRS | Line 11, Position 1: Found unparsable section: 'ALTER
                       | INDEX "PROJECTION_NAME_INDEX" RENA...'
==== summary ====
files:             1 violations:        2
clean files:       0 unclean files:     1
avg per file:   2.00 unclean rate:   100%
status:         FAIL
All Finished!
```

### How to reproduce

`sqlfluff lint --dialect postgres file.sql` on a SQL file with `ALTER INDEX`

### Dialect

postgres

### Version

Python 3.10.3

sqlfluff, version 0.11.2

### Configuration

```
[sqlfluff]
dialect = postgres
exclude_rules =
  # Unnecessary trailing whitespace
  L001,
  # Indentation not consistent with preceding line
  L003,
  # Files must end with a single trailing newline
  L009,
  # Keywords must be consistently upper case
  L010,
  # Line is too long
  L016,
  # Function name not immediately followed by parenthesis
  L017,
  # Unnecessary whitespace found
  L039,
  # Boolean/null literals must be consistently lower case
  L040,
  # Files must not begin with newlines or whitespace
  L050,
  # Unnecessary quoted identifier
  L059
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
